cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/cs/CsNetfilter.py

```diff
@@ -163,11 +164,14 @@ def apply_rules(self):
         chains.table_printout()
 
         # COMMIT all rules.
-        result = CsHelper.execute("iptables-restore < /tmp/rules.save")
-        if result:
-            logging.info("iptables-restore result: %s", result)
-        else:
+        p = Popen("iptables-restore < /tmp/rules.save", shell=True, stdout=PIPE, stderr=PIPE)
+        stdout, stderr = p.communicate()
+        if not stderr:
             logging.info("iptables-restore result: success!")
+        else:
+            print("iptables-restore failed, resulted in %s" % stderr)
```

Comment:
@borisroman The printed error shows up in the agent.log for easier troubleshooting in case the router hits the error while starting (and gets deleted).